File: server/models/Patient.js
const mongoose = require('mongoose');
const Appointment = require('./appointment');
const PatientSchema = new mongoose.Schema({
    fullname: {
        type: String,
        required: [true, "Fullname is Mandatory"],
        minlength: 6,
        trim: true
    },
    email: {
        type: String,
        required: [true, "Email is Mandatory"],
        unique: true,
        trim: true,
        match: /^\w+([-+.]\w+)*@((yahoo|gmail|outlook)\.com)$/
    },
    password: {
        type: String,
        required: [true, "Password is Required"],
        minlength: 6,
        trim: true
    },
    phone: {
        type: String,
        required: [true, "Phone is Required"],
        minlength: 10,
        trim: true
    },
    photo: {
        type: String,
    },
    gender: {
        type: String,
    },  
    age: {  
        type: String,
    },
    appointments: {
       
        type: [String],
        default: []
        
    },

}, { timestamps: true });

const Patient = mongoose.model('Patient', PatientSchema);
module.exports = Patient;

File: hospital/lib/models/Patient.dart
// Purpose: Patient model

import 'Appointment.dart';

class Patient {
  final String email;
  final String password;
  final String fullname;
  final String age;
  final String phone;
  final String photo;
  final String gender;
  final String id;
  final List<Appointment> appointments;
  final String token;

  Patient({
    required this.email,
    required this.password,
    required this.fullname,
    required this.age,
    required this.photo,
    required this.gender,
    required this.phone,
    required this.id,
    required this.appointments,
    required this.token,
  });
}
